### DISCUSSION TITLE
Please suggest an optimal value for queueLimit

### DISCUSSION THREAD

**Discussion by vinayk7** [moscajs/aedes#745] [Q&A] [answered]
[In the constructor](https://github.com/moscajs/aedes/blob/main/docs/Aedes.md#new-aedesoptions--new-aedesserveroptions) there's a parameter called `queueLimit` the explanation which is given I find is insufficient . Is the "buffering" limit like say we have 400 regular raspberry pi devices which I have connected to the main server via aedes on MQTT protocol so does the `queueLimit` say that by default only 42 unprocessed messages can be in queue and if another raspberry device happens to send (43rd message) it would get lost?
In those cases what should we do what will be the optimal value for the given setup of 400 devices, connectTimeout = 10secs, heartbeatInterval=5secs

**Comment by robertsLando** [comment, marked as answer]
> say that by default only 42 unprocessed messages can be in queue and if another raspberry device happens to send (43rd message) it would get lost?

It's a queue so it keeps last N messages

Set it to 0 if you don't want this to happen

**Comment by rajcse2005** [comment]
How to handle 1000000 clients or more then that in this. broker? 

Please help me on this issue?
